### PR TITLE
Made it possible to pass variables to the dev template via settings.

### PR DESCRIPTION
**Pull request by Naycon**
This pull request adds a setting for passing additional template variables to the nunjuck template the dev server uses. One use case for this is to be able to set secrets not suitable for version control in the shell environment for developers and pass them into the demo this way.

### CODE REVIEW

File: extensions/roc-package-web-component-dev/src/commands/nunjucks.js
Comment (dlmr) [inline extensions/roc-package-web-component-dev/src/commands/nunjucks.js]: Can we use the following syntax here?

``` js
{
  name,
  bundlePath: stats.script,
  ...template.variables
}
```

File: extensions/roc-package-web-component-dev/src/config/roc.config.meta.js
Comment (dlmr) [inline extensions/roc-package-web-component-dev/src/config/roc.config.meta.js]: Should probably be `An object...` here.

**Review by dlmr — CHANGES_REQUESTED** [review]

File: extensions/roc-package-web-component-dev/src/commands/nunjucks.js
Comment (dlmr) [inline extensions/roc-package-web-component-dev/src/commands/nunjucks.js]: Might it be better here to place all user defined variables under a single property? Like `userVariables` for example.

**Comment by dlmr** [comment]
Could we add how this new functionality can be used by updating the example?

**Review by dlmr — APPROVED** [review]

**Comment by Naycon** [comment]
Sure, will add an example on usage!

**Comment by dlmr** [comment]
Thanks! This looks good! 👍 

The only thing is that we now get warnings in the example for properties that Roc does not understand since Roc handles `settings` in a managed way. What I'm not trying to decide is how we should handle this case and similar ones that we will have in the future, we should definitely not get warnings at least. I see a couple of different options and I'm glad for feedback and other ideas.
1. Recommend that "custom" config values are used when using this type of generic objects.
2. Change the way the validation works to not get a warning here, for example if the object is empty from the extensions. The only problem with this is that it will not work great if another extension composes such a setting since it will then become managed and give warnings for the user provided values.
3. Use a new form of meta data to tell Roc that this object should be used as a generic value and not as a normal managed value. This is kinda the same thing as the custom alternative mentioned in 1. 

With this said I think we should hold off of merging this until we have decided the best approach.

**Comment by Naycon** [comment]
You're welcome! Makes sense @dlmr to hold off until a way forward regarding the config has been decided.

Regarding option 2, what do you mean by 'if another extension composes such a setting'?

**Comment by dlmr** [comment]
@Naycon An example of this would be if a `roc-package-web-component-react` would define a value on `settings.dev.template.customValues`.

**Comment by dlmr** [comment]
A suggestion is to update the `isObject` validator to allow it not validate nested keys using a new second argument:

``` js
isObject(true);
isObject(validator, true);
```

Using such a validator will not give warnings on mismatched keys and also allow other extensions to add keys. 

What do you think about this @Naycon?

**Comment by Naycon** [comment]
@dlmr - So if I define a new key in `roc-package-web-component-react` on that object with a validator, that would give me warnings for that setting if it is of the wrong type, but it would not give warnings if I define some custom keys as a user? If so, that would work. Maybe use an options object instead of just `true` as the final argument, to make it more future proof if you need to add similar solutions in the future?

**Comment by dlmr** [comment]
@Naycon Correct.

**Comment by dlmr** [comment]
I have now published a new version of `roc` that supports marking an object as _unmanaged_, `1.0.0-rc.14`. 

[Read more about the new option here.](https://github.com/rocjs/roc/blob/master/docs/Validators.md#isobject)

So you should be able to update the PR to use this new option and we can after that go ahead and merge this. 🙂

**Comment by Naycon** [comment]
Thanks @dlmr! I updated the PR to use the new isObject option and bumped the version of the roc dependency.

**Comment by dlmr** [comment]
Great stuff, thanks! 👍

**Comment by dlmr** [comment]
Released as `1.0.0-beta.3` 🎉

**Comment by Naycon** [comment]
🎉

**Pull request merged**
